File: Python/Create_Timer.py
import time


def countdown(total_seconds: int) -> None:
    while total_seconds >= 0:
        minutes, seconds = divmod(total_seconds, 60)
        print(f"\rTime left: {minutes:02d}:{seconds:02d}", end="")
        time.sleep(1)
        total_seconds -= 1
    print("\nTime is up!")


def main() -> None:
    try:
        seconds = int(input("Enter countdown time in seconds: ").strip())
        if seconds < 0:
            print("Please enter a non-negative number.")
            return
        countdown(seconds)
    except ValueError:
        print("Invalid input. Please enter a whole number.")


if __name__ == "__main__":
    main()

File: Javascript/Create_Timer.js
const readline = require('node:readline/promises');
const { stdin, stdout } = require('node:process');

const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

const countdown = async (totalSeconds) => {
  for (let seconds = totalSeconds; seconds >= 0; seconds -= 1) {
    const minutes = Math.floor(seconds / 60);
    const remaining = seconds % 60;
    process.stdout.write(
      `\rTime left: ${String(minutes).padStart(2, '0')}:${String(remaining).padStart(2, '0')}`,
    );
    await sleep(1000);
  }

  process.stdout.write('\nTime is up!\n');
};

const main = async () => {
  const rl = readline.createInterface({ input: stdin, output: stdout });

  try {
    const raw = (await rl.question('Enter countdown time in seconds: ')).trim();
    const seconds = Number(raw);

    if (!Number.isInteger(seconds) || seconds < 0) {
      console.log('Please enter a non-negative whole number.');
      return;
    }

    await countdown(seconds);
  } finally {
    rl.close();
  }
};

main().catch((error) => {
  console.error('Unexpected error:', error);
});
